provider/datagen/src/transform/cldr/displaynames/language.rs

```diff
@@ -61,43 +61,45 @@ impl DataProvider<LocaleDisplayNamesV1Marker> for crate::DatagenProvider {
 
 impl IterableDataProvider<LanguageDisplayNamesV1Marker> for crate::DatagenProvider {
     fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
-        Ok(self
-            .source
-            .cldr()?
-            .displaynames()
-            .list_langs()?
-            .filter(|langid| {
-                // The directory might exist without languages.json
-                self.source
-                    .cldr()
-                    .unwrap()
-                    .displaynames()
-                    .file_exists(langid, "languages.json")
-                    .unwrap_or_default()
-            })
-            .map(DataLocale::from)
-            .collect())
+        Ok(self.source.options.locales.filter_by_langid_equality(
```

Comment:
Discussion/Issue: I'm not really a fan of pulling this into each and every transformer. It seems very much prone to error.

I think I still prefer the design where the list of locales (or locale config) is provided directly to the `export` function instead of being in the datagen constructor.

Reply:
If `export` decided which locales to export, any key-specific logic (like for segmentation) would need to be in `export`. I think that's bad design because it would literally be an `if key == Marker::KEY { // choose locales }` for any number of keys, whereas with this design the logic for each key is in its provider implementation.

Reply:
Whether the locale config is provided to the constructor or `export` is independent of whether the actual logic is in `export` or the providers. Putting it in the constructor is just easier because passing it to `supported_locales` would require changing the public `IterableDataProvider` trait (or using internal mutability on `self` before calling it). I'm somewhat neutral as to where to provide it (I think `export` makes sense as well), but I think evaluating it in the provider impls is much cleaner.

Reply:
This seems not quite right to me but it's fine to land for now, and we can discuss further. https://github.com/unicode-org/icu4x/issues/3409